src/main/java/com/sap/cloudfoundry/client/facade/adapters/CloudFoundryClientFactory.java

```diff
@@ -25,31 +21,38 @@
 import org.immutables.value.Value;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.http.HttpStatus;
-
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.http.codec.json.Jackson2JsonDecoder;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.ExchangeStrategies;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.sap.cloudfoundry.client.facade.CloudException;
-import com.sap.cloudfoundry.client.facade.CloudOperationException;
 import com.sap.cloudfoundry.client.facade.Messages;
 import com.sap.cloudfoundry.client.facade.oauth2.OAuthClient;
 import com.sap.cloudfoundry.client.facade.rest.CloudSpaceClient;
 import com.sap.cloudfoundry.client.facade.util.CloudUtil;
 import com.sap.cloudfoundry.client.facade.util.JsonUtil;
 
+import io.netty.channel.ChannelOption;
 import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
 
 @Value.Immutable
 public abstract class CloudFoundryClientFactory {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CloudFoundryClientFactory.class);
 
-    static final HttpClient HTTP_CLIENT = HttpClient.newBuilder()
-                                                    .executor(Executors.newSingleThreadExecutor())
-                                                    .followRedirects(HttpClient.Redirect.NORMAL)
-                                                    .connectTimeout(Duration.ofMinutes(10))
-                                                    .build();
-
     private final Map<String, ConnectionContext> connectionContextCache = new ConcurrentHashMap<>();
 
+    private static final ObjectMapper MAPPER = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+                                                                 .configure(DeserializationFeature.UNWRAP_ROOT_VALUE, true);
```

Comment:
Does the unwrapping of the root value break the cf root response parsing? Or is this mapper not used when the response is interpreted as a String?

Reply:
No, the mapper and the decoder are not used when parsing Strings. It is only used when parsing the response to the respective java class (application logs response)